### PR TITLE
[Aleo ins.] Update commit and hash instructions.

### PR DESCRIPTION
**Pull request by acoglio**
Matches https://github.com/AleoHQ/snarkVM/pull/1656 and https://github.com/AleoHQ/snarkVM/pull/1647.

### CODE REVIEW

File: aleo.abnf [hunk 1]
```diff
@@ -354,9 +338,13 @@ is = is-op ws operand ws operand ws %s"into" ws register
...
 
-commit = commit-op ws operand ws operand ws %s"into" ws register
+commit = commit-op ws operand ws operand
+         ws %s"into" ws register
+         ws %s"as" ws ( %s"address" / %s"field" / %s"group" )
 
-hash = hash-op ws operand ws %s"into" ws register
+hash = hash-op ws operand
+       ws %s"into" ws register
+       ws %s"as" ws ( arithmetic-type / %s"address" )
```
Comment (bendyarm) [inline aleo.abnf:347]:
```suggestion
       ws %s"as" ws ( arithmetic-type / address-type )
```

**Review by bendyarm — APPROVED** [review]
LGTM.
Minor suggestion... I can see leaving it as %s"address" for consistency with commit-op,
but then why bother defining `address-type` at all?

File: aleo.abnf (hunk 1 above)
Comment (acoglio) [reply aleo.abnf]: @bendyarm Excellent idea. I've done that also for `field-type` and `group-type`. As explained in the commit message, while equivalent, it makes the role slightly more clear. And the grammar is slightly better factored this way.

**Review by bendyarm — APPROVED** [review]
LGTM

**Review by d0cd — APPROVED** [review]
LGTM!

**Pull request merged**